### PR TITLE
84962

### PR DESCRIPTION
**Pull request by nisanthyaganti9**
|Field|Value|
|-----|-----|
| Redmine Ticket Number | [84962](https://devops.dchbx.org/redmine/issues/84962) |
| App-Dev Road Map # | n/a |
| Start Date | n/a |
| Due Date | n/a  |
| App-Dev Priority | UP-Disqueued |
| Development Story Points | n/a |
| Peer Review Story Points | n/a |
| Ticket Author | @Ckellem |

### CODE REVIEW

File: app/controllers/api/v1/agencies_controller.rb [hunk 1]
```diff
@@ -54,20 +54,53 @@ def terminate
...
-    #   end
-    # rescue Exception => e
-    #   flash[:error] = "Role was not approved because " + e.message
-    # end
-    # redirect_to profiles_general_agencies_general_agency_profile_path(id: params[:staff][:profile_id])
+  def update_person
+    begin
+      query = Queries::People::AgencyStaffDetailQuery.new(current_user.person_id)
+      authorize query, :update_staff?
+      people = Person.where(first_name: /^#{update_person_params[:first_name]}$/i, last_name: /^#{update_person_params[:last_name]}$/i, dob: Date.strptime(update_person_params[:dob], "%Y-%m-%d").to_date)
```
Comment (antonioschaffert) [inline app/controllers/api/v1/agencies_controller.rb]: can we change `people` to `person`? We should be looking up the person by id, not first and last name, right? Or did something change?

Comment (nisanthyaganti9) [reply app/controllers/api/v1/agencies_controller.rb]: this is a check for an existing people in db with given details.

Comment (markgoho) [reply app/controllers/api/v1/agencies_controller.rb]: Right, this is a check to see if a person with that demographic information exists or not

File: app/controllers/api/v1/agencies_controller.rb [hunk 2]
```diff
@@ -54,20 +54,45 @@ def terminate
...
-    #   else
-    #     flash[:error] = "Role was not approved because " + @result
-    #   end
-    # rescue Exception => e
-    #   flash[:error] = "Role was not approved because " + e.message
-    # end
-    # redirect_to profiles_general_agencies_general_agency_profile_path(id: params[:staff][:profile_id])
+  def update_person
+    operation = Operations::UpdateStaff.new(update_person_params)
+    authorize operation, :update_staff?
```
Comment (antonioschaffert) [inline app/controllers/api/v1/agencies_controller.rb]: remove excess line returns.

File: app/controllers/api/v1/agencies_controller.rb [hunk 3]
```diff
@@ -54,20 +54,45 @@ def terminate
...
+    case operation.update_person
+    when :ok
+      render json: { status: "success", message: 'Succesfully updated!!' }, status: 200
```
Comment (antonioschaffert) [inline app/controllers/api/v1/agencies_controller.rb]: status codes are good, but can we make them constants?

Let's use 400 for bad request and not 404.

BAD_REQUEST = 400
OKAY_REQUEST = 200

**Comment by antonioschaffert** [comment]
Approved

File: app/controllers/api/v1/agencies_controller.rb [hunk 4]
```diff
@@ -44,30 +49,55 @@ def terminate
     authorize terminate_agency_staff, :terminate_agency_staff?
     case terminate_agency_staff.call
     when :ok
-      render json: { status: "success" }, status: 200
+      render json: { status: "success" }, status: OK_REQUEST_STATUS
```
Comment (bill-transue) [inline app/controllers/api/v1/agencies_controller.rb]: Let's use the builtin Rails HTTP status symbols. You can find them in the [Rails Guides](https://guides.rubyonrails.org/layouts_and_rendering.html#the-status-option).

**Pull request merged**